src/qibocal/protocols/characterization/signal_experiments/time_of_flight_readout.py

```diff
@@ -0,0 +1,179 @@
+
+    fitted_parameters: dict[QubitId, dict[str, float]]
+    """Raw fitting output."""
+
+
+TimeOfFlightReadoutType = np.dtype(
+    [("samples", np.float64), ("window_size", np.int64), ("sampling_rate", np.int64)]
+)
+
+
+@dataclass
+class TimeOfFlightReadoutData(Data):
+    """TimeOfFlightReadout acquisition outputs."""
+
+    data: dict[QubitId, npt.NDArray[TimeOfFlightReadoutType]] = field(
+        default_factory=dict
+    )
+    """Raw data acquired."""
+
+    def register_qubit(self, qubit, samples, window_size, sampling_rate):
+        """Store output for single qubit."""
+        ar = np.empty(samples.shape, dtype=TimeOfFlightReadoutType)
+        ar["samples"] = samples
+        ar["window_size"] = window_size
+        ar["sampling_rate"] = sampling_rate
+        self.data[qubit] = np.rec.array(ar)
+
+
+def _acquisition(
+    params: TimeOfFlightReadoutParameters, platform: Platform, qubits: Qubits
+) -> TimeOfFlightReadoutData:
+    """Data acquisition for time of flight experiment."""
+
+    sequence = PulseSequence()
+
+    RX_pulses = {}
+    ro_pulses = {}
+    for qubit in qubits:
+        RX_pulses[qubit] = platform.create_RX_pulse(qubit, start=0)
+        ro_pulses[qubit] = platform.create_qubit_readout_pulse(
+            qubit, start=RX_pulses[qubit].finish
+        )
+
+        sequence.add(ro_pulses[qubit])
+
+    results = platform.execute_pulse_sequence(
+        sequence,
+        options=ExecutionParameters(
+            nshots=params.nshots,
+            relaxation_time=params.relaxation_time,
+            acquisition_type=AcquisitionType.RAW,
+            averaging_mode=AveragingMode.CYCLIC,
+        ),
+    )
+
+    data = TimeOfFlightReadoutData()
+
+    # retrieve and store the results for every qubit
+    for qubit in qubits:
+        samples = results[ro_pulses[qubit].serial].magnitude
+        # store the results
+        data.register_qubit(qubit, samples, params.window_size, platform.sampling_rate)
+
+    return data
+
+
+def _fit(data: TimeOfFlightReadoutData) -> TimeOfFlightReadoutResults:
+    """Post-processing function for TimeOfFlightReadout."""
+
+    qubits = data.qubits
+    fitted_parameters = {}
+
+    for qubit in qubits:
+        qubit_data = data[qubit]
+        # Calculate moving average
+        # FIXME: This way of getting a parameter in the fit is saving it
+        # as many times as samples are on the data.
+        window_size = qubit_data.window_size[0]
+        i = 0
+        moving_average_deltas = []
+        last = None
+        while i < len(qubit_data) - window_size + 1:
+            window_average = (
+                np.sum(qubit_data[i : i + window_size].samples) / window_size
+            )
+            if last is None:
+                last = window_average
+            moving_average_deltas.append(window_average - last)
+            last = window_average
+            i += 1
+
+        max_average_change = moving_average_deltas.index(max(moving_average_deltas))
```

Comment:
Here I don't fully understand what is happening.
If you are interested in the max you should just use `max` or `np.max`. If you need the index you can use `np.argmax`.

Reply:
```suggestion
        max_average_change = np.argmax(moving_average_deltas) + 1
```
This should do the job I think. The `+1` could be necessary if you consider the comment above.

Reply:
I converted the list into a np.array and just argmax is enough, we don't need the +1